opm/parser/eclipse/EclipseState/Util/iorder_set.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+  Copyright 2019 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_IORDER_SET_HPP
+#define OPM_IORDER_SET_HPP
+
+#include <unordered_set>
+#include <vector>
+#include <string>
+#include <stdexcept>
+#include <iterator>
+
+namespace Opm {
+
+
+/*
+  Small class which implements an insert-ordered set. The set has an erase()
+  method which can be used to remove elements, otherwise the elements in the
+  container are immutable.
+*/
```

Comment:
It is not clear from the description how certain cases will be handled. What if I insert a value that was already in the container, will it be added again (i.e. the container is a pure sequence)? Or will nothing happen? Or will the ordering only be changed, with the value being now considered the newest value?

Reply:
See the tests. I'll add a brief comment.